### ISSUE TITLE
CA1823 triggered for field only used in attribute

### ISSUE THREAD

**Issue by 333fred** [dotnet/roslyn-analyzers#1191]
Given this class:

```C#
    [Guid(PackageGuid)]
    class FxCopAnalyzersPackage : Package
    {
        private const string PackageGuid = "4A41D270-A97F-4639-A352-28732FC410E4";
    }
```
CA1823 (Unused field) is reported for `PackageGuid`. This has a similar root cause to https://github.com/dotnet/roslyn-analyzers/issues/933, because IOperation doesn't expose an API for attributes currently. Related design issue is https://github.com/dotnet/roslyn/issues/18198.

**Comment by 333fred** [comment]
Dupe of https://github.com/dotnet/roslyn-analyzers/issues/933.

**Comment by 333fred** [comment]
This is not a dupe of #933.

**Comment by 333fred** [comment]
@mavasani, did we ever implement getting IOperations for code in attributes?

**Comment by mavasani** [comment]
GetOperation should work for syntax nodes in attributes, but I presume we do not return an operation block for attribute for operation block analyzers. For now, we might need to add a workaround in the analyzer to detect this.

**Comment by mavasani** [comment]
This should be fixed with the latest Roslyn fix that went into 15.5: Issue https://github.com/dotnet/roslyn/issues/23309, fixed with https://github.com/dotnet/roslyn/pull/23318. I will try to move the repo to newer Roslyn bits and verify the same.
